servant/src/Servant/API/TypeLevel.hs

```diff
@@ -241,18 +249,53 @@ We might try to factor these our more cleanly, but the type synonyms and type
 families are not evaluated (see https://ghc.haskell.org/trac/ghc/ticket/12048).
 -}
 
+-- ** Fragment
+
+class FragmentUnique api => OnlyOneFragment api
+
+-- | If fragment appeared in API endpoint twice, compile-time error would be raised.
+--
+-- >>> -- type FailAPI = Fragment Bool :> Fragment Int :> Get '[JSON] NoContent
+-- >>> instance OnlyOneFragment FailAPI
+-- ...
+-- ...Only one Fragment allowed per endpoint in api...
+-- ...
+-- ...In the instance declaration for...
+instance OnlyOneFragment (Verb m s ct typ)
+
+type family FragmentUnique api :: Constraint where
+  FragmentUnique (sa :<|> sb)       = Or (FragmentUnique sa) (FragmentUnique sb)
```

Comment:
Why `Or` and not `And`? 🤔

Reply:
- `Or` here represents difference between several endpoints. 
- The fragment uniqueness check should be applied only per particular API endpoint. Not to the whole API.

Reply:
```haskell
type FailApi =
  (Fragment "foo" Text :> Fragment "bar" Int :> Get '[JSON] Int)
  :<|> (Fragment "baz" String) :> Get '[JSON] String)
```

With `Or` your check will pass for this API because it passes for the second branch, ignoring the error in the first one.

Sorry if I'm getting something wrong :)

Reply:
Fixed.